pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -116,46 +118,53 @@ func GetDockercfgFile(path string) string {
 	return cfgPath
 }
 
-func readSpecificDockerConfigJSONFile(filePath string) bool {
+func readSpecificDockerConfigJSONFile(filePath string) error {
 	var contents []byte
 	var err error
 
 	if contents, err = ioutil.ReadFile(filePath); err != nil {
 		log.V(4).Infof("error reading file: %v", err)
-		return false
+		return err
 	}
 	return readDockerConfigJSONFileFromBytes(contents)
 }
 
-func readDockerConfigJSONFileFromBytes(contents []byte) bool {
+func readDockerConfigJSONFileFromBytes(contents []byte) error {
 	var cfgJSON credentialprovider.DockerConfigJSON
 	if err := json.Unmarshal(contents, &cfgJSON); err != nil {
-		log.V(4).Infof("while trying to parse blob %q: %v", contents, err)
-		return false
+		log.V(0).Infof("error trying to parse blob %q: %v", contents, err)
```

Comment:
Yeah, yesterday it looked like it ends up printing part of the secret, I don't think we can print the error itself by default. What about just printing "Error Deserializing, set env var X to value Y to see details (warning: This might leak the content of the secret"?

Reply:
yep I'll redo this part

Reply:
and to be clear, it was the `contents` piece here ... if we considered the error log that occurred here, `while trying to parse blob << the raw json pull secret >> ": illegal base64 data at input byte 1152`

The `contents` piece is `<< the raw json pull secret >>` and the err string is `illegal base64 data at input byte 1152`

I'll leave the err string

Reply:
I've also double checked `json.Unmarshal`  ... all the errors just note the byte array location like in the log above